src/allmydata/frontends/magic_folder.py

```diff
@@ -129,19 +234,19 @@ def _scan(self, localpath):
                 self.warn("WARNING: cannot backup symlink %s" % quote_local_unicode_path(childpath))
             elif isdir:
                 # process directories unconditionally
-                self._append_to_deque(childpath)
+                self._append_to_upload_deque(childpath)
 
                 # recurse on the child directory
                 self._scan(childpath)
             elif isfile:
-                must_upload = self._check_db_file(childpath)
-                if must_upload:
-                    self._append_to_deque(childpath)
+                is_uploaded = self._db_file_is_uploaded(childpath)
+                if not is_uploaded:
+                    self._append_to_upload_deque(childpath)
```

Comment:
See note at start of `_append_to_upload_deque` about whether we are checking the right condition for this case?